aten/src/ATen/native/ComplexHelper.h

```diff
@@ -4,12 +4,25 @@
 
 namespace at { namespace native {
 
-inline std::vector<int64_t> computeStrideForViewAsReal(IntArrayRef oldstride) {
-  auto res = oldstride.vec();
-  for(size_t i = 0; i < res.size(); i++) {
-    res[i] = res[i] * 2;
+// View tensor with new dtype, storage offset, sizes and strides 
+inline Tensor view_tensor(
+    const Tensor &tensor, ScalarType dtype,
+    int64_t offset, IntArrayRef sizes, IntArrayRef strides) {
+  Storage storage = tensor.storage();
+  auto new_tensor = detail::make_tensor<TensorImpl>(
+      std::move(storage), tensor.key_set(), scalarTypeToTypeMeta(dtype));
+  auto * impl = new_tensor.unsafeGetTensorImpl();
+  impl->set_storage_offset(offset);
+  impl->set_sizes_and_strides(sizes, strides);
+  return new_tensor;
+}
+
+inline DimVector computeStrideForViewAsReal(IntArrayRef oldstride) {
+  DimVector res(oldstride.size() + 1);
+  for(size_t i = 0; i < oldstride.size(); i++) {
+    res[i] = oldstride[i] * 2;
   }
-  res.emplace_back(1);
+  res.back() = 1;
```

Comment:
just curious if there's a reason why `back()` is more preferable thank `emplace_back()`

Reply:
@peterbell10 pre-allocated the size of `res` to avoid a (possible) reallocation

Reply:
Yeah, `emplace_back` inserts a new element while `back` is just a shorthand for indexing the last element.